Lib/ctypes/test/test_pep3118.py

```diff
@@ -17,14 +17,47 @@ def normalize(format):
     format = format.replace(OTHER_ENDIAN, THIS_ENDIAN)
     return re.sub(r"\s", "", format)
 
+def replace_type_codes(format):
+    # Replace ctypes type codes with struct/PEP3118 codes. They are
+    # different in the struct module "standard" size mode.
+    code_map = {
+        "h": (c_short, {2: 'h', 4: 'l', 8: 'q'}),
+        "H": (c_ushort, {2: 'H', 4: 'L', 8: 'Q'}),
+        "i": (c_int, {2: 'h', 4: 'i', 8: 'q'}),
+        "I": (c_uint, {2: 'H', 4: 'I', 8: 'Q'}),
+        "l": (c_long, {4: 'l', 8: 'q'}),
+        "L": (c_ulong, {4: 'L', 8: 'Q'}),
+        "q": (c_longlong, {8: 'q'}),
+        "Q": (c_ulonglong, {8: 'Q'}),
+        "?": (c_bool, {1: '?', 2: 'H', 4: 'L', 8: 'Q'}),
+        "d": (c_double, {8: 'd'}),
+        "g": (c_longdouble, {sizeof(c_longdouble): 'g'}),
+    }
+
+    replacements = {}
+    for ctypes_code in code_map.keys():
+        # Since some types may be aliases to each other, look up the
+        # struct code based on the actual ctypes type
+        tp = code_map[ctypes_code][0]
+        struct_code = code_map[tp._type_][1][sizeof(tp)]
```

Comment:
I must say I don't understand it either anymore. It may be leftover from refactoring.
Since the tests seem to pass, I now simplified it.